airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/registry.py

```diff
@@ -93,9 +132,12 @@ def generate_and_persist_registry(
 # Registry Generation
 
 
-@asset(required_resource_keys={"slack", "registry_directory_manager", "latest_oss_registry_entries_file_blobs"}, group_name=GROUP_NAME)
+@asset(
+    required_resource_keys={"slack", "registry_directory_manager", "latest_oss_registry_entries_file_blobs", "latest_metrics_gcs_blob"},
+    group_name=GROUP_NAME,
+)
 @sentry.instrument_asset_op
-def persisted_oss_registry(context: OpExecutionContext) -> Output[ConnectorRegistryV0]:
+def persisted_oss_registry(context: OpExecutionContext, latest_connnector_metrics: dict) -> Output[ConnectorRegistryV0]:
```

Comment:
For my own understanding: the reason we pass this in as a param instead of pulling it from the context directly is because of the amount of [processing that needs to be done on it ](https://github.com/airbytehq/airbyte/pull/38681/files#diff-054e388c4ad695ea0affcdc80aac1d4da2409a4b462cb11ae27d9ff172cf6ce3R58-R62), and we'd prefer to do that asynchronously and store it in an asset rather than calculating it when we're trying to persist?

Reply:
Resource usage/control is a small benefit.

But the reason I did this was to
1. Have a separate node in the dag for discoverablity (For future devs, so they have an FYI we take data from here and pipe it into the registry, when using Dagster)
2. Start an asset where we may do heavier processing and want finer grained control over when to materialize
3. Have a separate area of our code base for this